africanus/averaging/bda_avg.py

```diff
@@ -757,72 +758,132 @@ def nb_bda_impl(
+        max_fov=3.0,
+        decorrelation=0.98,
+        time_bin_secs=None,
+        min_nchan=1,
+    ):
+        # Merge flag_row and flag arrays
+        flag_row = merge_flags(flag_row, flag)
+
+        meta = bda_mapper(
+            time,
+            interval,
+            antenna1,
+            antenna2,
+            uvw,
+            chan_width,
+            chan_freq,
+            max_uvw_dist,
+            flag_row=flag_row,
+            max_fov=max_fov,
+            decorrelation=decorrelation,
+            time_bin_secs=time_bin_secs,
+            min_nchan=min_nchan,
+        )
 
-    row_avg = row_average(
-        meta,
-        antenna1,
-        antenna2,
-        flag_row,  # noqa: F841
-        time_centroid,
-        exposure,
-        uvw,
-        weight=weight,
-        sigma=sigma,
-    )
+        row_avg = row_average(
+            meta,
+            antenna1,
+            antenna2,
+            flag_row,  # noqa: F841
+            time_centroid,
+            exposure,
+            uvw,
+            weight=weight,
+            sigma=sigma,
+        )
 
-    row_chan_avg = row_chan_average(
-        meta,  # noqa: F841
-        flag_row=flag_row,
-        visibilities=visibilities,
-        flag=flag,
-        weight_spectrum=weight_spectrum,
-        sigma_spectrum=sigma_spectrum,
-    )
+        row_chan_avg = row_chan_average(
+            meta,  # noqa: F841
+            flag_row=flag_row,
+            visibilities=visibilities,
+            flag=flag,
+            weight_spectrum=weight_spectrum,
+            sigma_spectrum=sigma_spectrum,
+        )
 
-    # Have to explicitly write it out because numba tuples
-    # are highly constrained types
-    return AverageOutput(
-        meta.map,
-        meta.offsets,
-        meta.decorr_chan_width,
-        meta.time,
-        meta.interval,
-        meta.chan_width,
-        meta.flag_row,
-        row_avg.antenna1,
-        row_avg.antenna2,
-        row_avg.time_centroid,
-        row_avg.exposure,
-        row_avg.uvw,
-        row_avg.weight,
-        row_avg.sigma,
-        # None,  # chan_data.chan_freq,
-        # None,  # chan_data.chan_width,
-        # None,  # chan_data.effective_bw,
-        # None,  # chan_data.resolution,
-        row_chan_avg.visibilities,
-        row_chan_avg.flag,
-        row_chan_avg.weight_spectrum,
-        row_chan_avg.sigma_spectrum,
-    )
+        # Have to explicitly write it out because numba tuples
+        # are highly constrained types
+        return AverageOutput(
+            meta.map,
+            meta.offsets,
+            meta.decorr_chan_width,
+            meta.time,
+            meta.interval,
+            meta.chan_width,
+            meta.flag_row,
+            row_avg.antenna1,
+            row_avg.antenna2,
+            row_avg.time_centroid,
+            row_avg.exposure,
+            row_avg.uvw,
+            row_avg.weight,
+            row_avg.sigma,
+            # None,  # chan_data.chan_freq,
```

Comment:
Is there a reason why chan_freq is not returned?

Reply:
Good question. Part of the reason is that it technically becomes a (row, chan) dimension as the frequencies and channel widths can also vary per row. This sort of data is available meta object, higher up in this namedtuple construction. There's a chan_width member with (row, chan) dimensionality. However, there isn't a `chan_freq` member. Do you need it? I suppose it wouldn't be necessary if one were only doing time BDA.

Reply:
Ah there's already an issue

- https://github.com/ratt-ru/codex-africanus/issues/263

Reply:
I should add that when doing time only BDA you shoudn't need uvw to be expanded as rowchan. This actually uses quite a bit of memory. It's not a train smash for now but it might be worth adding a time only BDA function in the future that doesn't do this